### DISCUSSION TITLE
how to write a fallback for method not allow?

### DISCUSSION THREAD

**Discussion by 799189288** [tokio-rs/axum#2824] [Q&A] [answered]
### Summary

there is a fallback to handle path not match, but where is the function to handle method not allow if the path matched

### axum version

0.7.1

**Comment by 799189288** [comment]
i find fallback on get or post method. but if i want to add a global fallback for method. what should i do

**Reply by mladedav** [reply]
I think [this issue](https://github.com/tokio-rs/axum/issues/2251) is what you're asking for?

**Reply by 799189288** [reply]
so you are working on it?  i think if i only set a get method for a path and i use post to request that path.it's appropriate to set a fallback as 404 as well rather than return with empty and 405 status.  will this feature be added in the next patch release?

**Reply by mladedav** [reply]
I am not working on it and I don't know whether someone else is actively working on it, but I don't think so based on the timeline of the comments. I just remembered that I've seen an issue like this.

Feel free to comment on the issue with your use-case, e.g. if you need the fallback be the same, if you don't want to send the `Allow` headers at all even if the path matched or anything you might think is relevant.

I guess there is still some design work to be done but there aren't enough people and this just wasn't a priority.

**Comment by FelixZY** [comment, marked as answer]
This functionality was actually merged [here](https://github.com/tokio-rs/axum/pull/2903) and is now documented [here](https://github.com/Lachstec/axum/blob/main/axum/src/docs/routing/method_not_allowed_fallback.md).
